packages/pipeline-editor/src/PipelineSubmissionDialog.tsx

```diff
@@ -33,7 +34,9 @@ interface IState {
 export class PipelineSubmissionDialog extends React.Component<IProps, IState> {
   state = {
     displayedRuntimeOptions: new Array<IRuntime>(),
-    selectedRuntimePlatform: this.props.schema[0] && this.props.schema[0].name,
+    selectedRuntimePlatform:
+      this.props.runtime?.name ||
+      (this.props.schema[0] && this.props.schema[0].name),
```

Comment:
If avoidable, we should try to not use props inside state

Reply:
What would be the better way to go about setting the state to that value (`this.props.runtime?.name`) by default?

Reply:
I would try to construct it outside of state and leave state for only storing the information that comes from some sort of event (user input / network request / etc...)

Reply:
something like this: https://codesandbox.io/s/sharp-waterfall-iymcv?file=/src/App.tsx